src/mpcCoreKit.ts

```diff
@@ -871,12 +872,22 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
     return tssNonce;
   }
 
-  private async setupTkey(importTssKey?: string): Promise<void> {
+  private async setupTkey(providedImportTssKey?: string): Promise<void> {
     if (!this.state.postBoxKey) {
       throw CoreKitError.userNotLoggedIn();
     }
     const existingUser = await this.isMetadataPresent(this.state.postBoxKey);
+    let importTssKey = providedImportTssKey;
     if (!existingUser) {
+      if (!importTssKey && !this.options.useDkg) {
+        if (this.keyType === KeyType.ed25519) {
+          importTssKey = bytesToHex(randomBytes(32));
+        } else if (this.keyType === KeyType.secp256k1) {
+          importTssKey = generateFactorKey().private.toString("hex", 64);
```

Comment:
this should use `secp256k1.genKeyPair` instead. this is not a factor key and if we would ever change the factor key type, this would cause unexpected errors.